src/transform/src/canonicalize_mfp.rs

```diff
@@ -71,6 +71,7 @@ impl CanonicalizeMfp {
                     .push_filters_through_map(&map, &mut filter, mfp.input_arity, all_errors);
                 if !pushdown.is_empty() {
                     *relation = relation.take_dangerous().filter(pushdown);
+                    crate::fusion::filter::Filter.action(relation);
```

Comment:
Originally I thought that this call was redundant because this call calls `canonicalize_predicates`, which is already called on line 68, but I think that there are cases where the predicates that end up in `pushdown` need to be re-canonicalized. Can you add at least one unit test that tests this line?